### PR TITLE
Device settings

### LINKED ISSUE #242: Add device rotation to (advanced?) settings
In firmware 2.1.1 we introduced permanent screen rotation (https://github.com/trezor/trezor-core/issues/162). That means we've disabed the "Swipe to rotate" gesture and user can rotate the screen only using `trezorctl set-display-rotation {north, south, west, east}`.

Would be nice to add this feature to Wallet, so users do not need `trezorctl`.

### PR DESCRIPTION
**Pull request by mroz22**
Tinkering about device settigs, view this PR as a initial impulse for discussion. 

resolves #242

### CODE REVIEW

**Comment by slowbackspace** [comment]
It is not in our OKR for Q3. closing... (joke 😀)

**Comment by mroz22** [comment]
Added some tests. Its not 100% but reaching it might be postponed to the next iteration in which we will implement users custom images. I would say it is good enough now. 

Maybe merge after sharing of static assets is in develop https://github.com/trezor/trezor-suite/pull/387

File: packages/suite/src/utils/suite/tests/homescreen.test.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+] as const;
+
+interface Fixture {
+    img: typeof imgHashFixtures[number]['img'];
+    model: typeof imgHashFixtures[number]['model'];
+    hexHash: typeof imgHashFixtures[number]['hexHash'];
+}
+
+describe('homescreen', () => {
+    describe('imageDataToHomescreen', () => {
```
Comment (vladimirvolek) [inline packages/suite/src/utils/suite/tests/homescreen.test.ts:60]: should be "it" or "test"

File: packages/suite/src/views/suite/settings/index.tsx [hunk 2]
```diff
@@ -1,14 +1,307 @@
-import React from 'react';
+/* eslint-disable @typescript-eslint/camelcase */
+import React, { useEffect, useState } from 'react';
 import styled from 'styled-components';
+import { Input, Button, P, Icon, Select } from '@trezor/components';
+import { AppState } from '@suite-types/index';
+import { resolveStaticPath } from '@suite-utils/nextjs';
+import elementToHomescreen from '@suite/utils/suite/elementToHomescreen';
+
+const IMAGE_NAMES_T1 = [
```
Comment (vladimirvolek) [inline packages/suite/src/views/suite/settings/index.tsx]: move to constants

File: packages/suite/src/utils/suite/tests/homescreen.test.ts (hunk 1 above)
Comment (mroz22) [reply packages/suite/src/utils/suite/tests/homescreen.test.ts:60]: Depends on what you want. Here I want to organize tests in 3 levels deep structure like this

image

This way it is used on many other places, for example: src/utils/suite/tests/validators.test.ts, src/utils/suite/tests/router.test.ts ...

File: packages/suite/src/views/suite/settings/index.tsx (hunk 2 above)
Comment (mroz22) [reply packages/suite/src/views/suite/settings/index.tsx]: As you wish. I am not a furious advocate of code splitting at all costs, especially in cases when variables might be used only in place where they are defined and nowhere else, which is the case. On the other hand, settings/index.tsx is going to grow with time, so moving these variables to constants might be desirable.

**Comment by mroz22** [comment]
Lets discuss how we are going to call actions on device. See my last commit. At least we need some prop on suite reducer that will block ui (buttons..) and unblock them in moment action was completed.

**Comment by mroz22** [comment]
consulted with szymon. added SUITE.UI_LOCK action. need to resolve some ts. not ready yet.

**Comment by mroz22** [comment]
fuh, please review.

**Review by slowbackspace — APPROVED** [review]

**Pull request merged**